ui/management/commands/retranscode_collections.py

```diff
@@ -0,0 +1,271 @@
+            created_after = self._parse_date(options["created_after"], "created-after")
+            queryset = queryset.filter(created_at__gte=created_after)
+
+        if options.get("created_before"):
+            created_before = self._parse_date(
+                options["created_before"], "created-before"
+            )
+            queryset = queryset.filter(created_at__lte=created_before)
+
+        return queryset
+
+    def _parse_date(self, date_string, argument_name):
+        """Parse date string and return timezone-aware datetime"""
+        try:
+            # Try parsing as full datetime first
+            parsed_date = parse_datetime(date_string)
+            if parsed_date:
+                # If no timezone info, assume UTC
+                if timezone.is_naive(parsed_date):
+                    parsed_date = timezone.make_aware(parsed_date, timezone.utc)
+                return parsed_date
+
+            # Try parsing as date only (YYYY-MM-DD)
+            try:
+                parsed_date = datetime.strptime(date_string, "%Y-%m-%d")
+                # For "created-after", use start of day; for "created-before", use end of day
+                if argument_name == "created-before":
+                    parsed_date = parsed_date.replace(
+                        hour=23, minute=59, second=59, microsecond=999999
+                    )
+                return timezone.make_aware(parsed_date, timezone.utc)
+            except ValueError:
+                pass
+
+            raise CommandError(
+                f"Invalid date format for --{argument_name}: '{date_string}'. "
+                f"Use YYYY-MM-DD or YYYY-MM-DD HH:MM:SS format."
+            )
+        except Exception as e:
+            raise CommandError(
+                f"Error parsing --{argument_name} date '{date_string}': {str(e)}"
+            )
+
+    def _get_videos_count(self, collections):
+        """Count videos for retranscoding"""
+        result = collections.aggregate(total=Count("videos"))
+        return result["total"] or 0
+
+    def _show_dry_run_details(self, collections, options):
+        """Show detailed information about what would be retranscoded"""
+        self.stdout.write("\n" + "=" * 80)
+        self.stdout.write(
+            "DRY RUN - Collections and videos that would be retranscoded:"
+        )
+        self.stdout.write("=" * 80)
+
+        # Show applied filters
+        if options.get("created_after"):
+            self.stdout.write(f"Filter: Created after {options['created_after']}")
+        if options.get("created_before"):
+            self.stdout.write(f"Filter: Created before {options['created_before']}")
+        if options.get("created_after") or options.get("created_before"):
+            self.stdout.write("")
+
+        total_videos = 0
+        for collection in (
+            collections.select_related("owner")
+            .annotate(video_count=Count("videos"))
+            .iterator()
+        ):
```

Comment:
This will avoid N+1 queries on line 221 and 231:

```suggestion
        for collection in (
            collections.select_related("owner")
            .prefetch_related("edx_endpoints", "videos")
            .annotate(video_count=Count("videos"))
            .iterator()
        ):
```